module/zfs/vdev_raidz.c

```diff
@@ -213,20 +207,21 @@ vdev_raidz_cksum_finish(zio_cksum_report_t *zcr, const void *good_data)
 			 */
 			for (x = 0; x < rm->rm_firstdatacol; x++) {
 				bad_parity[x] = rm->rm_col[x].rc_abd;
-				rm->rm_col[x].rc_gdata =
-				    zio_buf_alloc(rm->rm_col[x].rc_size);
 				rm->rm_col[x].rc_abd =
-				    abd_get_from_buf(rm->rm_col[x].rc_gdata,
-				    rm->rm_col[x].rc_size);
+				    rm->rm_col[x].rc_gdata =
+				    abd_alloc_for_io(rm->rm_col[x].rc_size,
```

Comment:
Maybe switch to `abd_alloc_sametype(rm->rm_col[x].rc_abd)` because not all raidz rec code can work with abd, and parity columns are currently linear buffers.